app/components/sorted_table_header_link_component.rb

```diff
@@ -0,0 +1,139 @@
+  # @param [Symbol] table_sort_field_key the current sort field of the table.
+  # @param [Symbol] table_sort_order_key the current sort order of the table.
+  # @return [SortedTableHeaderLinkComponent::LinkMaker] a LinkMaker object whose link method returns a SortedTableHeaderLinkComponent with prepopulated fields.
+  def self.link_maker(params:, table_sort_field_key: :sort_field, table_sort_order_key: :sort_order)
+    raise ArgumentError.new("Unpermitted params") unless params.permitted?
+    self::LinkMaker.new(
+      params: params,
+      table_sort_field_key: table_sort_field_key,
+      table_sort_order_key: table_sort_order_key
+    )
+  end
+
+  # @param [String] column_title The text of the link; the column title as displayed to the user.
+  # @param [String] sort_field the new sort field that will be used if a user clicks the link.
+  # @param [String] table_sort_field the current sort field of the table.
+  # @param [String] table_sort_order the current sort order of the table.
+  # @param [String] table_sort_field_key the key in params identifying the current sort order of the table.
+  # @param [String] table_sort_order_key the key in params identifying the current sort order of the table.
+  # @param [Hash] other_params any and all extra paramaters to add to the link.
+  def initialize(column_title:, sort_field:,
+    table_sort_field:, table_sort_order:,
+    table_sort_field_key: :sort_field, table_sort_order_key: :sort_order,
+    other_params:)
+    unless column_title.present? && sort_field.present?
+      raise ArgumentError.new("Missing arguments: column_title or sort field") 
+    end
+    unless table_sort_order.in? ['asc', 'desc']
+      raise ArgumentError.new("Sort order needs to be asc or desc") 
+    end
+    @column_title = column_title
+    @sort_field = sort_field
+    @table_sort_field = table_sort_field
+    @table_sort_order = table_sort_order
+    @table_sort_field_key = table_sort_field_key
+    @table_sort_order_key = table_sort_order_key
+    @other_params = other_params
+  end
+
+  def call
+    link_to sort_column_title, sort_column_path
+  end
+
+
+  def sort_column_path
+    url_param_hash = @other_params || {}
+    url_param_hash[@table_sort_field_key] = @sort_field
+
+    # SORT ORDER:
+    # We support only two possible sort orders: ascending and descending.
+    # We are not managing default per-column sort orders.
+    if @sort_field == @table_sort_field
+      # IF the column whose title we want to display is the *same*
+      # as the column the table is currently sorted by,
+      # THEN, IF the user clicks on this title,
+      #  we reverse the direction of the sort.
+      url_param_hash[@table_sort_order_key] = reversed_sort_order
+    else
+      # For columns OTHER than the one the table is currently sorted by,
+      # clicking the title of the column will sort the table by that column,
+      # in an arbitrary order.
```

Comment:
I _think_ I understand, but what do you mean by "arbitrary"? WE only have two kinds, asc and desc, don't we know for sure which one we are doing? With "asc" being the default?